dex/networks/eth/txdata.go

```diff
@@ -55,33 +143,45 @@ func ParseInitiateData(calldata []byte) ([]swapv0.ETHSwapInitiation, error) {
 		return nil, fmt.Errorf("expected first arg of type []swapv0.ETHSwapInitiation but got %T", args[0].value)
 	}
 
-	toReturn := make([]swapv0.ETHSwapInitiation, 0, len(initiations))
+	// This is done for the compiler to ensure that the type defined above and swapv0.ETHSwapInitiation
+	// are the same, other than the tags.
+	if len(initiations) > 0 {
+		_ = swapv0.ETHSwapInitiation(initiations[0])
```

Comment:
When doing the type assertion, it must be the exact same type, including the name of the type. If it's anonymous struct, asserting with a named struct, even if it contains the same fields/tags, will not work:
https://go.dev/play/p/hJO53Ptj0Af

> But also doesn't the assertion `args[0].value.([]struct...` do the same thing effectively?

What do you mean? The reason for this is so the compiler will tell us to change this if we change the solidity code. This line won't do that.

Reply:
> including the name of the type

Ughhh, yes according to the spec "A **defined** type is always different from any other type. Otherwise, two types are identical if ... [the rest of the rules about fields and tags]"  https://go.dev/ref/spec#Type_identity

This all feels like such a hack using an anonymous type.  I think I know the answer, but again, I would like to ask the go-ethererum developers why there is no exported interface for inspecting the call data or why the exported type generated by abigen is not also json-tagged.

Reply:
> I think I know the answer, but again, I would like to ask the go-ethererum developers why there is no exported interface for inspecting the call data or why the exported type generated by abigen is not also json-tagged.

What do you mean? You don't thing they'll do it?